### PR TITLE
Prompt are you still on duty

### LINKED ISSUE #25: Prompt are you still on duty? 10 hours after going on duty

### PR DESCRIPTION
**Pull request by akaktus**
[-] Implemented notification after 4 hours after closing app
[-] Implemented notification after 10 hours after starting duty.
Fixes #25, #26

### CODE REVIEW

File: o-fish-ios/Helpers/NotificationManager.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    func removeAllNotification() {
+        notificationCenter.removeAllDeliveredNotifications()
+        notificationCenter.removeAllPendingNotificationRequests()
+    }
+
+    func requestNotification(_ hours: Int, afterClosing: Bool = true) {
+        let seconds = TimeInterval(hours * 3600)
+        let content = UNMutableNotificationContent()
+        let localizedString = afterClosing ? NSLocalizedString("It has been %@ hours after closing the application", comment: "") :
```
Comment (kojiba) [inline o-fish-ios/Helpers/NotificationManager.swift:40]: make two separate methods this notifications

File: o-fish-ios/Helpers/NotificationManager.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        notificationCenter.delegate = delegate
+    }
+
+    func removeAllNotification() {
+        notificationCenter.removeAllDeliveredNotifications()
+        notificationCenter.removeAllPendingNotificationRequests()
+    }
+
+    func requestNotification(_ hours: Int, afterClosing: Bool = true) {
+        let seconds = TimeInterval(hours * 3600)
```
Comment (kojiba) [inline o-fish-ios/Helpers/NotificationManager.swift:38]:
```
let secondsInHour = 3600 
let seconds = TimeInterval(hours * secondsInHour)
```

File: o-fish-ios/SceneDelegate.swift [hunk 3]
```diff
@@ -14,6 +14,7 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
     var settings = Settings.shared
     var locationHelper = LocationHelper()
     var imageCache = ImageCache()
+    let hoursAfterClosing = 4
```
Comment (kojiba) [inline o-fish-ios/SceneDelegate.swift:17]: dependency inject with constants:
```
let hoursAfterClosing = Constants.hoursAfterClosing
...

struct Constants {
...
let notifactionHoursAfterClosing = 4
let notifactionHoursAfterStarting = ...
...
```

File: o-fish-ios/Views/PatrolBoat/PatrolBoatView.swift [hunk 4]
```diff
@@ -17,6 +17,9 @@ class DutyState: ObservableObject {
         }
     }
 
+    static let shared = DutyState()
+    let hoursAfterStarting = 10
```
Comment (kojiba) [inline o-fish-ios/Views/PatrolBoat/PatrolBoatView.swift:21]: dependency inject with constants

File: o-fish-ios/Views/PatrolBoat/PatrolBoatView.swift [hunk 5]
```diff
@@ -34,6 +37,11 @@ class DutyState: ObservableObject {
         }
         dutyChangeViewModel.user = user
         dutyChangeViewModel.status = onDuty ? .onDuty : .offDuty
+        if onDuty {
```
Comment (kojiba) [inline o-fish-ios/Views/PatrolBoat/PatrolBoatView.swift:40]: is this a correct place when the duty synchronise?

File: o-fish-ios/Views/PatrolBoat/PatrolBoatView.swift [hunk 6]
```diff
@@ -17,6 +17,9 @@ class DutyState: ObservableObject {
         }
     }
 
+    static let shared = DutyState()
+    let hoursAfterStarting = 10
+
     init() {
```
Comment (kojiba) [inline o-fish-ios/Views/PatrolBoat/PatrolBoatView.swift:23]: make all constructors private to use only 
`DutyState.shared `

File: o-fish-ios/Helpers/NotificationManager.swift [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+    func requestAutorization() {
+        notificationCenter.requestAuthorization(options: [.alert, .sound, .badge]) { (didAllow, error) in
+            if !didAllow {
+                print("User has declined notifications")
+            }
+
+            if let error = error {
+                print(error.localizedDescription)
+            }
```
Comment (kojiba) [inline o-fish-ios/Helpers/NotificationManager.swift:24]: should we add extra code to handle notification send when user declined ?

File: o-fish-ios/Views/PatrolBoat/PatrolBoatView.swift (hunk 5 above)
Comment (akaktus) [reply o-fish-ios/Views/PatrolBoat/PatrolBoatView.swift:40]: It's called before duty state will be saved (next line). So, I guess, It's correct place.

File: o-fish-ios/Helpers/NotificationManager.swift (hunk 7 above)
Comment (akaktus) [reply o-fish-ios/Helpers/NotificationManager.swift:24]: We can add alert with propose for user turns on notification, but I don't know, where and when should be this alert.

Comment (kojiba) [reply o-fish-ios/Helpers/NotificationManager.swift:24]: Regarding Apple policy we prohibited to use such alerts

Comment (akaktus) [reply o-fish-ios/Helpers/NotificationManager.swift:24]: I can add such alerts after changing on/off duty state (from off to on) with proposal turn on notification and link to settings. How many times should I use this alert?

File: o-fish-ios/Helpers/Constants.swift [hunk 8]
```diff
@@ -12,4 +12,6 @@ struct Constants {
     static let realmAppId = "your-realm-app-id"
 
     static let DefaultLocation = CLLocationCoordinate2D(latitude: 51.506520923981554, longitude: -0.10689139236939127)
+    static let hoursAfterClosing = 4
```
Comment (kojiba) [inline o-fish-ios/Helpers/Constants.swift:15]: add enum Notifications
and move configs there

```
enum Constants
...
    enum Notifications {
        static let hoursAfterClosing = 4
        ...
    }
...
}
```

**Pull request merged**